pcuic/theories/README.md

```diff
@@ -2,133 +2,289 @@
+| [PCUICInstConv]     | Stability of cumulativities by instantiation
+| [PCUICWeakeningEnvConv] | Stability of cumulativities by global environment extension
+| [PCUICUnivSubstitutionConv] | Stability of cumulativities by substitution of universe variables
+| [PCUICClosedConv] | Helper lemmas on the closedness predicate
+| [PCUICOnFreeVarsConv] | Helper lemmas for renamings and free variables
+
+[PCUICRenameConv]: ./Conversion/PCUICRenameConv.v
+[PCUICWeakeningConv]: ./Conversion/PCUICWeakeningConv.v
+[PCUICInstConv]: ./Conversion/PCUICInstConv.v
+[PCUICWeakeningEnvConv]: ./Conversion/PCUICWeakeningEnvConv.v
+[PCUICUnivSubstitutionConv]: ./Conversion/PCUICUnivSubstitutionConv.v
+[PCUICClosedConv]: ./Conversion/PCUICClosedConv.v
+[PCUICOnFreeVarsConv]: ./Conversion/PCUICOnFreeVarsConv.v
+
+
+## Stability of Typing
+
+| File                | Description                                               |
+|---------------------|-----------------------------------------------------------|
+| [PCUICGuardCondition]| Axioms on the stability of the guard condition 
+| [PCUICRenameTyp]    | Stability of typing by renaming
+| [PCUICWeakeningTyp] | Stability of typing by extension of the local context (special case of renaming)
+| [PCUICInstTyp]      | Stability of typing by instatiation
+| [PCUICSubstitution] | Stability of typing by substitution (derived as a special case of stability by instantiation)
+| [PCUICWeakeningEnvTyp] | Stability of typing by extension of the global environment
+| [PCUICUnivSubstitutionTyp] | Stability of typing by substitution of universe levels
+| [PCUICConversionTyp] | Stability of typing by cumulativity of the local context
+
+[PCUICGuardCondition]: ./PCUICGuardCondition.v
+[PCUICRenameTyp]: ./Typing/PCUICRenameTyp.v
+[PCUICWeakeningTyp]: ./Typing/PCUICWeakeningTyp.v
+[PCUICInstTyp]: ./Typing/PCUICInstTyp.v
+[PCUICSubstitution]: ./PCUICSubstitution.v
+[PCUICWeakeningEnvTyp]: ./Typing/PCUICWeakeningEnvTyp.v
+[PCUICUnivSubstitutionTyp]: Typing/PCUICUnivSubstitutionTyp.v
+[PCUICConversionTyp]: Typing/PCUICContextConversionTyp.v
+
+
+## Metatheoretical properties of typing
+
+| File              | Description
+|-------------------|----------------------------------------------------
+| [PCUICContexts]     | Various properties on contexts
+| [PCUICArities]    | Properties on lists of terms
+| [PCUICSpine]      | Properties on spines (lists of terms) relating to typing, substitutions…
+| [PCUICInductives] | Technical typing and conversion lemmatas related to inductive definitions
+| [PCUICValidity] | Every term `A` such that `Γ ⊢ t : A` is a type
+| [PCUICInductiveInversion] | Typing properties and inversions for inductive types, constructors and projections.
+| [PCUICAlpha] | Typing does not depend on names
+| [PCUICSR]    | Subject reduction: typing is preserved by reduction
+| [PCUICPrincipality] | Existence of a least type for any typable term
```

Comment:
Yes, now this claim is only derived as a consequence. But anyway the proof of principality is not really needed anymore, because composing correctness + completeness of inference, together with uniqueness of inferred types gives principality "for free".

Reply:
Oh you don't rely on this result to prove it? In any case I'll mark this as resolved.